### PR TITLE
[Test] Remove dead code from ExceptionSerializationTests

### PR DESCRIPTION
**Pull request by cbuescher**
The `ignore` set contains entries of type Class<?>, but the check is performed
on Path objects. This always returns false so is useless currently. Looking at
the first commit of this test shows that this already was there from the start of this
test, so this behaviour seems to have never resulted in anything. My current guess is is 
can be removed.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/ExceptionSerializationTests.java [hunk 1]
```diff
@@ -133,11 +133,7 @@ public void testExceptionRegistration()
 
             @Override
             public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                Path next = pkgPrefix.resolve(dir.getFileName());
-                if (ignore.contains(next)) {
-                    return FileVisitResult.SKIP_SUBTREE;
-                }
-                pkgPrefix = next;
```
Comment (atorok) [inline server/src/test/java/org/elasticsearch/ExceptionSerializationTests.java]: Are you sure it's ok to remove this line ? Line 189 reads it.

Comment (cbuescher) [reply server/src/test/java/org/elasticsearch/ExceptionSerializationTests.java]: My reasoning is this: nothing in this method in this current form changes "next" (the contains() call should have no side effects), so next is still "pkgPrefix.resolve(dir.getFileName())" here, which is what I use to set pkgPrefix in the following line. Does that make sense? I'm not fully sure what the ignore.contains() check was supposed to do at some point, but I think it doesn't do anything now.

**Review by alpar-t — APPROVED** [review]
My bad, I missed that next was really just in-lined. 
Looks good.

**Comment by cbuescher** [comment]
@atorok thanks for the review, will merge to master

**Pull request merged**